canopen_ros2_controllers/test/test_canopen_proxy_controller.cpp

```diff
@@ -0,0 +1,280 @@
+// Copyright (c) 2022, Stogl Robotics Consulting UG (haftungsbeschränkt)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "test_canopen_proxy_controller.hpp"
+
+#include <limits>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+class CanopenProxyControllerTest : public CanopenProxyControllerFixture<TestableCanopenProxyController>
+{
+};
+
+// When there are many mandatory parameters, set all by default and remove one by one in a
+// parameterized test
+TEST_P(CanopenProxyControllerTestParameterizedParameters, one_parameter_is_missing)
+{
+  SetUpController();
+
+  ASSERT_EQ(controller_->on_configure(rclcpp_lifecycle::State()), NODE_ERROR);
+}
+
+// TODO(anyone): the new gtest version after 1.8.0 uses INSTANTIATE_TEST_SUITE_P
+INSTANTIATE_TEST_SUITE_P(
+  MissingMandatoryParameterDuringConfiguration, CanopenProxyControllerTestParameterizedParameters,
+  ::testing::Values(
+    std::make_tuple(std::string("joint"), rclcpp::ParameterValue(std::string({})))));
+
+TEST_F(CanopenProxyControllerTest, joint_names_parameter_not_set)
+{
+  SetUpController(false);
+
+  ASSERT_TRUE(controller_->joint_name_.empty());
+
+  ASSERT_EQ(controller_->on_configure(rclcpp_lifecycle::State()), NODE_ERROR);
+
+  ASSERT_TRUE(controller_->joint_name_.empty());
+}
+
+TEST_F(CanopenProxyControllerTest, all_parameters_set_configure_success)
+{
+  SetUpController();
+
+  ASSERT_TRUE(controller_->joint_name_.empty());
+
+  ASSERT_EQ(controller_->on_configure(rclcpp_lifecycle::State()), NODE_SUCCESS);
+
+  // TODO(livanov93): check if it is string
+  // hardcoded as it should be "joint1"
+  ASSERT_THAT(controller_->joint_name_, testing::HasSubstr("joint"));
```

Comment:
Wouldn't this be better?
```suggestion
  ASSERT_THAT(controller_->joint_name_, joint_name_);
```